src/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinHttpHandler.cs

```diff
@@ -785,7 +785,7 @@ private void EnsureSessionHandleExists(WinHttpRequestState state)
                 // Specify an HTTP server.
                 connectHandle = Interop.WinHttp.WinHttpConnect(
                     _sessionHandle,
-                    state.RequestMessage.RequestUri.Host,
+                    state.RequestMessage.RequestUri.HostNameType == UriHostNameType.IPv6 ? $"[{state.RequestMessage.RequestUri.IdnHost}]" : state.RequestMessage.RequestUri.IdnHost,                    
```

Comment:
@caesar1995 @davidsh @rmkerr i added `[]` for IPv6 because i get `'The URL is invalid'` for `http://[::1234]` address

Reply:
> because i get 'The URL is invalid' for http://[::1234] address.

Could you elaborate on this? What part of the code will complain "The URL is invalid"? What test did you do?